client/json-rpc/rpc_service.c

```diff
@@ -47,10 +49,166 @@
 #define RPC_METHOD_NOT_FOUND -32601
 #define RPC_INVALID_PARAMS -32603
 #define RPC_INTERNAL_ERROR -32693
+#define RPC_WHITE_ADDR_LEN          64
+#define RPC_WHITE_MAX               8
+
 
 const uint32_t RPC_SERVER_PORT = 7677; //default http json-rpc port 7677
 
-static int send_response(struct xdag_rpc_connection * conn, char *response) {
+struct item {
+    struct item *prev, *next;
+    struct in_addr addr;
+};
+
+struct item *g_rpc_white_host = NULL;
+
+static int rpc_host_addr_check_ipv4(const char *host){
+
+  int n[4];
+  char c[4];
+
+  if (!host){
+    return 0;
+  }
+
+  if (sscanf(host, "%d%c%d%c%d%c%d%c",
+             &n[0], &c[0], &n[1], &c[1],
+             &n[2], &c[2], &n[3], &c[3]) == 7)
+
+  {
+    int i;
+    for(i = 0; i < 3; ++i)
+      if (c[i] != '.')
+        return 0;
+    for(i = 0; i < 4; ++i)
+      if (n[i] > 255 || n[i] < 0)
+        return 0;
+    return 1;
+  } else
+    return 0;
+}
+
+static int rpc_white_host_check(struct sockaddr_in peeraddr){
+
+  struct item *node = NULL ;
+    
+  if (!g_rpc_white_host){
+        return 1;
+  }
+
+  LL_FOREACH(g_rpc_white_host,node)
+  {
+    if (node->addr.s_addr == peeraddr.sin_addr.s_addr){
+        return 1;
+    }
+  }
+
+  return 0;
+}
+
+
+static int rpc_white_host_add(const char *host){
```

Comment:
@g2fly99 Need check if the IP is already added before add to list to avoid duplicate IPs.